packages/composable-controller/src/ComposableController.ts

```diff
@@ -63,8 +104,6 @@ export class ComposableController extends BaseController<
   ComposableControllerState,
   ComposableControllerMessenger
 > {
-  readonly #controllers: ControllerList = [];
```

Comment:
Alternatively, we could keep the `#controllers` field and add something like the following to `#updateChildController`.

```ts
if (!this.state[name]) {
  this.#controllers.push(controller)
}
```

I prefer removing the field because it makes it unambiguous that child controllers aren't meant to be added, and their state and subscriptions aren't meant to be mutated after instantiation. 

Also, having ComposableController depend on mutable internal state would introduce complications for #3627.